File: app/src/main/java/org/deiverbum/app/ui/fragments/CalendarioFragment.java
package org.deiverbum.app.ui.fragments;

import android.os.Bundle;
import android.view.ContextMenu;
import android.view.LayoutInflater;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.DatePicker;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;

import org.deiverbum.app.R;
import org.deiverbum.app.databinding.FragmentCalendarioBinding;
import org.deiverbum.app.utils.Utils;

import java.util.Calendar;
import java.util.Locale;
import java.util.Objects;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class CalendarioFragment extends Fragment {
    private FragmentCalendarioBinding binding;
    private String mDate;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = FragmentCalendarioBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        DatePicker datePicker = binding.datePicker;

        Calendar calendar = Calendar.getInstance();
        datePicker.init(calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH), (datePicker1, year, month, dayOfMonth) -> {
            datePicker1.performLongClick();
            mDate = String.format(Locale.getDefault(),"%d%02d%02d",year,
                    month+1,dayOfMonth);
            updateActionBar();

        });
        this.registerForContextMenu(datePicker);
        return root;
    }

    private void updateActionBar() {
        ActionBar actionBar = ((AppCompatActivity) requireActivity()).getSupportActionBar();
        Objects.requireNonNull(actionBar).setSubtitle(Utils.getTitleDate(mDate));
    }

    public void onCreateContextMenu(@NonNull ContextMenu menu, @NonNull View v, ContextMenu.ContextMenuInfo menuInfo) {
        MenuInflater inflater = requireActivity().getMenuInflater();
        inflater.inflate(R.menu.calendario_menu, menu);
        super.onCreateContextMenu(menu, v, menuInfo);
    }

    public boolean onContextItemSelected(MenuItem item) {
        if (item.getItemId() == 0) {
            return true;
        }
        Bundle bundle = new Bundle();
        bundle.putString("FECHA", mDate);
        NavController navController = NavHostFragment.findNavController(this);
        navController.navigate(item.getItemId(),bundle);
        return true;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/org/deiverbum/app/core/presentation/calendar/CalendarioFragment.kt
package org.deiverbum.app.core.presentation.calendar

import android.os.Bundle
import android.view.ContextMenu
import android.view.ContextMenu.ContextMenuInfo
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.DatePicker
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.NavHostFragment
import dagger.hilt.android.AndroidEntryPoint
import org.deiverbum.app.R
import org.deiverbum.app.databinding.FragmentCalendarioBinding
import org.deiverbum.app.util.Utils.formatDate
import java.util.Calendar
import java.util.Locale

@AndroidEntryPoint
class CalendarioFragment : Fragment() {
    private var binding: FragmentCalendarioBinding? = null
    private var mDate: String? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        super.onCreate(savedInstanceState)
        binding = FragmentCalendarioBinding.inflate(inflater, container, false)
        val root: View = binding!!.root
        val datePicker = binding!!.datePicker
        val loc = Locale("es", "ES")
        val calendar = Calendar.getInstance()
        datePicker.init(
            calendar[Calendar.YEAR],
            calendar[Calendar.MONTH],
            calendar[Calendar.DAY_OF_MONTH]
        ) { datePicker1: DatePicker, year: Int, month: Int, dayOfMonth: Int ->
            datePicker1.performLongClick()
            mDate = String.format(
                loc, "%d%02d%02d", year,
                month + 1, dayOfMonth
            )
            updateActionBar()
        }
        registerForContextMenu(datePicker)
        return root
    }

    private fun updateActionBar() {
        val actionBar = (requireActivity() as AppCompatActivity).supportActionBar
        actionBar?.subtitle =
            formatDate(mDate, "yyyyMMdd", "d '-' MMMM yyyy")
    }

    override fun onCreateContextMenu(menu: ContextMenu, v: View, menuInfo: ContextMenuInfo?) {
        val inflater = requireActivity().menuInflater
        inflater.inflate(R.menu.calendario_menu, menu)
        super.onCreateContextMenu(menu, v, menuInfo)
    }

    override fun onContextItemSelected(item: MenuItem): Boolean {
        if (item.itemId == 0) {
            return true
        }
        val bundle = Bundle()
        bundle.putInt("FECHA", mDate!!.toInt())
        val navController = NavHostFragment.findNavController(this)
        navController.navigate(item.itemId, bundle)
        return true
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }
}